source/SIL.AppBuilder.Portal.Frontend/src/sockets/data-socket.tsx

```diff
@@ -14,122 +9,189 @@ import {
 
 import { isTesting } from '~/env';
 
-import { TransformOrOperations } from '@orbit/data';
-import { Observable, Subscription } from 'rxjs';
-import { map } from 'rxjs/operators';
+import { TransformOrOperations, Source } from '@orbit/data';
+import { Subscription } from 'rxjs';
 import JSONAPISource from '@orbit/jsonapi';
 import Store from '@orbit/store';
 
 import { DataClient } from './clients';
 
 import { dataToLocalCache } from '~/data/orbitjs-operations-support/serialize-from-api';
 
-import { useMemoIf } from '~/lib/hooks';
-
-import { DataHub } from './clients/data';
-
 import { useAuth } from '~/data/containers/with-auth';
 
-const mockClient: DataClient = {
-  connection: {
-    connectionState$: {
-      subscribe() {},
-    },
-    send() {},
-    invoke() {},
-    hubConnection: { state: undefined },
-  },
-  start() {},
-  stop() {},
-  dataStore: undefined,
-  hubFactory: undefined,
-  hubName: undefined,
-  onData$$: undefined,
-  onDataReceived: undefined,
-  onReceive: undefined,
-  subscription$$: {},
-};
-
-export default function LiveDataManager({ children }) {
+export default function LiveDataManger({ children }) {
   const { dataStore, sources } = useOrbit();
   const { isLoggedIn, auth0Id } = useAuth();
 
-  const hubFactory = useMemoIf(() => new HubConnectionFactory(), !isTesting, [isLoggedIn]);
-  const dataClient =
-    useMemoIf(() => new DataClient(hubFactory, dataStore), !isTesting, [isLoggedIn]) || mockClient;
-  const subscriptions = useMemo(() => [], [isLoggedIn]);
+  return (
+    <LiveDataManagerSocketManager
+      {...{
+        dataStore,
+        sources,
+        isLoggedIn,
+        auth0Id,
+      }}
+    >
+      {children}
+    </LiveDataManagerSocketManager>
+  );
+}
+
+interface LiveDataProps {
+  dataStore: Store;
+  isLoggedIn: boolean;
+  auth0Id?: string;
+  sources: {
+    [sourceName: string]: Source;
+  };
+  children: (providedArgs: {
+    socket: DataClient;
+    pushData: (transforms: TransformOrOperations) => Promise<JSONAPIOperationsPayload>;
+    subscriptions: string[];
+    isConnected: boolean;
+    connectionState: ConnectionState;
+  }) => React.ReactNode;
+}
+
+interface State {
+  isConnected: boolean;
+  connectionState: ConnectionState;
+}
+
+// using a class instead of a series of functional components and hooks
```

Comment:
For this particular use case, react hooks are an absolute terrible fit for comprehension